lib/populator.js

```diff
@@ -271,24 +271,30 @@ Populator.prototype._checkForDuplicateEvent = function(knownKeys, trackedEntityI
     }
 
     if (!body.events || body.events.length === 0) {
-      return next(null, trackedEntityInstanceID);
+      return next(null, trackedEntityInstanceID, false);
     }
-
-    next(new Error('Duplicate event'));
-  });
+    
+    if (this._options.duplicateStageID) {
+      return next(null, trackedEntityInstanceID, true);
+    }
+    
+    next(new Error('Duplicate Event'));
```

Comment:
Looking at this now I suggest you revert changing the case of the 'e' in the error message so that we don't have to worry about the mediator breaking. Using proper version numbers and shrinkwrapping should prevent any issues, but since this change to the message really isn't necessary I think it's better to leave it out.